sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/UnchangedLocalVariablesShouldBeConst.cs

```diff
@@ -0,0 +1,261 @@
+
+            var3 += 1;
+
+            return false;
+        }
+
+        enum Colors { Red, Blue, Green, Nice };
+
+        public void Test_MultipleTypes(int arg)
+        {
+            char cc = 'c'; // Noncompliant
+            int intVar = 1; // Noncompliant
+            const int constIntVar = 1;
+            int intVar2 = 1 + constIntVar; // Noncompliant
+            long longVal = 1; // Noncompliant
+            short shortVal = 1; // Noncompliant
+            double doubleVal = 1; // Noncompliant
+            System.Int16 int16Val = 1; // Noncompliant
+            ulong ulongVal = 1; // Noncompliant
+            int bufferSize = 512 * 1024; // Noncompliant
+            decimal val = 123.456m; // Noncompliant
+            double val2 = .1d; // Noncompliant
+
+            Colors c1 = Colors.Nice; // Noncompliant
+            Colors c2 = (Colors)1; // Noncompliant
+            Colors c3 = 0.0; // Noncompliant
+
+            string str1 = ""; // Noncompliant
+            string str2 = null; // Noncompliant
+            string str3 = nameof(arg); // Noncompliant
+            string str4 = typeof(int).Name;
+            string str5 = "a" + "b"; // Noncompliant
+            string str6 = @"a" + $"b";
+            string str7 = System.Environment.NewLine;
+            string str8 = String.Empty;
+
+            object obj1 = null; // Noncompliant
+            object obj2 = new object();
+            object obj3 = TestUtils.GetObj();
+            object obj4 = 1;
+
+            IEnumerable<string> enumerable = null; // Noncompliant
+
+            //dynamic value = 42; // Compliant. Const only works with null.
+            //long? notExpected = 100; // Compliant. Const only works with null.
+            //TestUtils x = 2; // Compliant (type with implicit conversion). Const only works with null.
+        }
+
+        public void Test_Property
+        {
+            get
+            {
+                int a = 1; // Noncompliant
+            }
+            set
+            {
+                int a = 1;
+                int b = 1; // Noncompliant
+                a = 2;
+            }
+        }
+
+        public static Tests operator +(Tests a, Tests b)
+        {
+            int i = 1; // Noncompliant
+            return null;
+        }
+
+        public Tests()
+        {
+            int a = 1; // Noncompliant
+        }
+
+
+        public void Test_FalseNegatives()
+        {
+            int a;
+            a = 1;
+        }
+
+        public byte this[int idx]
+        {
+            get
+            {
+                var pos = 0;
+                var seen = 0;
+
+                seen += 1;
+                pos += 1;
+
+                return 1;
+            }
+        }
+
+
+        // invalid code
+        public void Test_TypeThatCannotBeConst(int arg)
+        {
+            System.Random random = 1;
+        }
+
+        // invalid code
+        public void (int arg)
+        {
+            int intVar = 1;
+        }
+    }
+}
```

Comment:
Are there any test cases using arrays?

The following examples shouldn't trigger the rule as they can't be made _const_:
_int[] x = { 1, 2, 3 };
int[] y = new int[] { 1, 2, 3 };
Exception[] z = { };_